src/lando/main/models/repo.py

```diff
@@ -104,10 +114,15 @@ def get_mapping(cls) -> dict[str, "Repo"]:
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        if self.is_hg_repo:
-            self.hg = HgRepo(self.system_path or self.get_system_path())
-        else:
-            self.hg = None
+
+    def get_scm(self) -> AbstractScm:
```

Comment:
Making it a `@property` could be a problem, as they are not cached. This would result in returning a fresh instance everytime, which could be confusing if we expect state to be retained between uses (and we'd potentially be leaking memory).

That said, there is some awkward logic between the `scm` and the `_scm` property, that I think we can make cleaner. The main issue is that the `scm` property maps to the text column in the DB, so we can't readily swap it out.

Thinking more about it, we could have an `scm_interface` `@property`, but we'd need to do some memoisation ourselves, so we return the pre-existing instance, if any, e.g.,

```python
@property
def scm_interface(self):
    if not self._scm:
      if self.scm == SCM_HG:
          self._scm = HgSCM(self.path)
    return self._scm
```

... which is pretty much exactly the what the `get_scm` already does, but it's not a `@property`. So, yeah, could be made one, but we'd need to retain the `_scm` attribute. I'm not sure we'd benefit much here.